### PR TITLE
GOLD-168

### PR DESCRIPTION
**Pull request by kgmyatthu**

### CODE REVIEW

**Comment by kgmyatthu** [comment]
Most stuff are straight forward but special review  needed for acceptance of Node[] | NodeWithRank[] in tellBinary() Peronsally this is fine since networking function only really cares for ip/port. 

Another place is poqo-send-vote vs spreadAppliedVoteHashReq are virtually the same, so ser/de poqo-send-vote is essentially a wrapper around it. Thought about being redundant but I don't see a reason atm yet. @arhamj

File: src/state-manager/TransactionConsensus.ts [hunk 1]
```diff
@@ -1063,6 +1065,111 @@ class TransactionConsenus {
...
+              Array.from(nodesToSendTo),
+              false,
+              4,
+              queueEntry.acceptedTx.txId
+            )
+            nestedCountersInstance.countEvent(`processing`, `forwarded final data to storage nodes`)
+          }
+          if (!queueEntry.hasSentFinalReceipt) {
+            if (logFlags.verbose) this.mainLogger.debug(`POQo: received data & receipt for ${queueEntry.logID} starting receipt gossip`)
+            queueEntry.poqoReceipt = readableReq.receipt
```
Comment (jintukumardas) [inline src/state-manager/TransactionConsensus.ts:1150]: payload (receipt) verification is missing

File: src/state-manager/TransactionConsensus.ts [hunk 2]
```diff
@@ -1063,6 +1065,111 @@ class TransactionConsenus {
...
+              queueEntry.transactionGroup,
+              false,
+              4,
+              readableReq.finalState.txid,
+              '',
+              true
+            )
+            queueEntry.hasSentFinalReceipt = true
+          }
+        } finally {
```
Comment (jintukumardas) [inline src/state-manager/TransactionConsensus.ts:1166]: catch block is missing. Not sure if this is nested try catch.

File: src/state-manager/TransactionConsensus.ts [hunk 3]
```diff
@@ -1200,6 +1307,34 @@ class TransactionConsenus {
...
+          const readableReq = deserializePoqoSendVoteReq(stream)
+          const queueEntry = this.stateManager.transactionQueue.getQueueEntrySafe(readableReq.txid)
+          if (queueEntry == null) {
+            /* prettier-ignore */ nestedCountersInstance.countEvent('poqo', 'poqo-send-vote: no queue entry found')
+            return
+          }
+          const collectedVoteHash = readableReq as AppliedVoteHash
+          // We can reuse the same function for POQo
+          this.tryAppendVoteHash(queueEntry, collectedVoteHash)
+        } finally {
```
Comment (jintukumardas) [inline src/state-manager/TransactionConsensus.ts:1330]: catch block is missing.

**Comment by kgmyatthu** [comment]
are these merged? @Glitch18 please close it with remark.

**Comment by Glitch18** [comment]
Yes, these were merged in a different PR. #172 
This can be closed

**Pull request closed without merging**